[ master < T0797 DX] Rewrite triggers reference guide and how to. (#478)
memgraph/docs: docs/reference-guide/triggers.md

> sidebar_label: Triggers

Reviewer: Please add related How-to guide box! :)

sidebar_label: Triggers
---

> [![Related -How-to](https://img.shields.io/static/v1?label=Related&message=How-to&color=blue&style=for-the-badge)](/how-to-guides/how-to-setup-triggers.md)

Reviewer: Unfortunately, we can't have these labels as a first thing on the page because when we paste links, the short description turns out to be markdown code, so we try to first have a little intro, then add this label... so just switch up the text and the label

( BEFORE | AFTER ) COMMIT
EXECUTE openCypherStatements
```
As you can see from the format, you can choose on what object event needs to happen , on `()` node or `-->` relationship. After that you can define on what type of event you what to execute the trigger `CREATE`, `UPDATE` or `DELETE`. After the `EXECUTE` is series of cypher statements you want to execute.

> An example of the trigger would be:

Reviewer: an example of a trigger would be

> The query may seem complex, so let's break it down:

Reviewer: I am not a fan of this flow where we have a general format, then give an example and explain it, then explain the general stuff again

I would prefer:
1. general format
2. Explain general format and each possible case
3. Give an example and explain the example

But if you don't agree I can live with this :) But this sentence "You can find detailed explanations for each part of the trigger [further down](#trigger-name)." is bugging me... why not explain it to me right now, that's why I'm here

Author: I agree, revamped it a bit.

* `CREATE TRIGGER exampleTrigger`: This statement creates the trigger.
* `ON UPDATE AFTER COMMIT EXECUTE`: This statement specifies what kind of event
  should activate the execution of trigger. This one will be triggered for every update
  operation and the query below will be executed after the update event has been
  committed.

> * `UNWIND updatedObjects AS updatedObject`: If multiple objects were updated,

Reviewer: I would add a part of the query where you define the name just so we are 100% clear what we are talking about

### Predefined variables
Statements that a trigger executes can contain certain predefined variables

> which contain information about the event that triggered it. Values of predefined variables are determined 
> by database transaction, multiple creations, updates or deletes that are part of single transaction
> will be in predefined variables. 

Reviewer 2: ```suggestion
which contain information about the event that triggered it. Values of predefined variables are determined 
by database transaction, multiple creations, updates or deletes that are part of a single transaction. 
```

> | ON --> DELETE | deletedEdges |

Reviewer: Should we maybe add some examples?

Author: Yes, but I do not know some logical example for that at the moment, so I will add it to tasks and update with example later.
